src/main/java/seedu/address/model/person/predicates/TagsContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,47 @@
+package seedu.address.model.person.predicates;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+
+/**
+ * Tests that a {@code Person}'s {@code Tags} matches any of the keywords given.
+ */
+public class TagsContainsKeywordsPredicate implements Predicate<Person> {
+
+    private final List<String> keywords;
+
+    public TagsContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return person.getTags().stream().map(tag -> keywords.stream()
+                    .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(tag.tagName, keyword)))
+                    .reduce(false, (x, y) -> x || y);
```

Comment:
Very nice use of streams and lambda functions. Perhaps the readability of this complex stream manipulation could be improved with better used of newlines and indentation.
For example, the outer stream(), map() and reduce() should be on the same indentation level, while the inner stream() and anyMatch() can be on a separate indentation level.

Reply:
Thank you for pointing that out, I completely agree! I have made the changes.